src/xdocs/checks/annotation/annotationlocation.xml

```diff
@@ -0,0 +1,281 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<document xmlns="http://maven.apache.org/XDOC/2.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/XDOC/2.0 https://maven.apache.org/xsd/xdoc-2.0.xsd">
+  <head>
+    <title>AnnotationLocation</title>
```

Comment:
Why isn't file named `annotationLocation.xml`?

Reply:
I think there wasn't any discussion about naming those. The names in the xdocs folder are mixed - `writingchecks.xml` ,`property_types.xml`, `writingfilters.xml`, `beginning_development.xml`. I just went with that but anything is fine by me as long as we stay consistent.

Reply:
@romani @nrmancuso

Reply:
>annotationLocation.xml

It is url naming,  java convention on naming should not be applied.
I completely satisfied with all lowercased.

Reply:
I also prefer hyphen, but let’s do underscore to be consistent

Reply:
If we started to make breaking compatibility for our urls, we can migrate to `-`. Users do not care much, they can migrate their links.

But this unrelated to this PR .

Reply:
Done. Changed them to use `_`

Reply:
Opened an issue for the inconsistent names at https://github.com/checkstyle/checkstyle/issues/13159. I'm leaving the names lowercase as is.